test/e2e/flowaggregator_test.go

```diff
@@ -102,21 +102,45 @@ const (
 )
 
 func TestFlowAggregator(t *testing.T) {
-	skipIfDualStackCluster(t)
-	data, isIPv6, err := setupTestWithIPFIXCollector(t)
+	data, v4Enabled, v6Enabled, err := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
-	podAIP, podBIP, podCIP, svcB, svcC, err := createPerftestPods(data, isIPv6)
+	podAIP, podBIP, podCIP, err := createPerftestPods(data)
 	if err != nil {
-		t.Fatalf("Error when creating perftest pods and services: %v", err)
+		t.Fatalf("Error when creating perftest pods: %v", err)
 	}
-	// Wait for the Service to be realized.
-	time.Sleep(3 * time.Second)
 
+	if v4Enabled {
+		svcB, svcC, err := createPerftestServices(data, false)
+		if err != nil {
+			t.Fatalf("Error when creating IPv4 perftest services: %v", err)
+		}
+		// Wait for the Service to be realized.
+		time.Sleep(3 * time.Second)
+
+		runTests(t, data, podAIP, podBIP, podCIP, svcB, svcC, false)
+		deletePerftestServices(data)
```

Comment:
maybe use a `defer` statement after the call to `createPerftestServices`

same below

Reply:
Thanks, addressed.